src/components/Main.tsx

```diff
@@ -423,6 +425,16 @@ const EdgeTabs = () => {
 // -------------------------------------------------------------------------
 
 const EdgeAppStack = () => {
+  const [countryCode, setCountryCode] = React.useState<string | undefined>()
+
+  useAsyncEffect(
+    async () => {
+      setCountryCode(await getCountryCodeByIp())
```

Comment:
The `getCountryCodeByIp` function does a network call. We don't want to do a bunch of network calls as we navigate the app - it would be better to just do one call at boot-up, and then use the cached value everywhere. We already do this with `getFirstOpenInfo` - if we've successfully loaded the file, it just returns the cached value instead of loading it again.

Now that we are using the `getCountryCodeByIp` function in so many places, I believe we need the same treatment to avoid a major performance regression.

Reply:
Yeah I was trying to come up with a new method to do it once and pass it around somehow without success, those are good ideas